### PR TITLE
Add notification email to discourse setup

### PR DESCRIPTION
**Pull request by pfaffman**
This is primarily to add ability to set SMTP_DOMAIN and NOTIFICATION_EMAIL in `discourse-setup`. 

@xfalcox asked for this:

- Add a new question to discourse-setup: “What is the email where notifications should come from?” or something like that.
- User replies, let’s say noreply@example.com.
- You split on the @ and set example.com to SMTP_DOMAIN.
- You use the full value and set the notification_email.

This does the above, and will add the required fields to reasonably old existing config files.

My last pull request was broken if it was re-run for existing sites, so this time I have added some basic smoke tests. As part of that I also added some command line switches to `--skip-rebuild` (that works) and `--skip-connection-test` (so we can run this on a host that isn't on the internet). (I also added `--two-container` and deprecated the rather ugly `2container` option).

The tests just run `discourse-setup` and test that values got set or changed as expected. I actually caught a couple bugs!

I also changed the port check to use `nc` rather than `netstat`, which is not included in some distributions.

### CODE REVIEW

**Review by xfalcox — APPROVED** [review]

**Comment by xfalcox** [comment]
Can you please rebase @pfaffman

**Comment by pfaffman** [comment]
OK. I think I got it this time.

**Comment by xfalcox** [comment]
It still can be rebased on top of master due to conflicts @pfaffman

**Comment by pfaffman** [comment]
Sigh, but it said there were no conflicts when I clicked the button...

I'll try again.

**Comment by pfaffman** [comment]
image

Did you not like my changes to check_port()?

**Comment by pfaffman** [comment]
Now I'm more confused. Is this approved now? Or do I need to try to rebase again?

**Comment by pfaffman** [comment]
Do I need to do anything else? It looks like we're just waiting for approval from another reviewer, right?

**Comment by xfalcox** [comment]
I did not change anything, the branch has 28 commits from you, but it needs to be rebased so it can be cleanly merged into master.

I can take a look into doing the rebase myself tomorrow.

**Comment by pfaffman** [comment]
image

Hmm. I thought that I'd successfully rebased and pushed.  I'll try once more to rebase.

**Comment by pfaffman** [comment]
If you can do the rebase in a few minutes, please do so. I spent an hour or so the other day, apparently to no end. 

If it's not quick for you, I'll probably just make sure to do a fresh pull, create a new branch, copy the files over there and try another PR.

**Comment by xfalcox** [comment]
Close in favor of #527

**Pull request closed without merging**
